test/hotspot/jtreg/compiler/intrinsics/base64/TestBase64.java

```diff
@@ -177,4 +236,139 @@ private static void assertEqual(Object result, Object expect) {
             }
         }
     }
+
+    private static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        if (len % 2 != 0) {
+                throw new InternalError("Internal test error: HEXASCII source file line doesn't contain an even number of characters: " + Integer.toString(len));
+        }
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            byte upperNibble = (byte)Character.digit(s.charAt(i), 16);
+            byte lowerNibble = (byte)Character.digit(s.charAt(i + 1), 16);
+            if (upperNibble == -1) {
+                throw new InternalError("Internal test error: HEXASCII source file contains non-hex character: " + s.charAt(i) + " at pos: " + Integer.toString(i));
+            }
+            if (lowerNibble == -1) {
+                throw new InternalError("Internal test error: HEXASCII source file contains non-hex character: " + s.charAt(i) + " at pos: " + Integer.toString(i + 1));
+            }
+            data[i / 2] = (byte)((upperNibble << 4) + lowerNibble);
+        }
+        return data;
+    }
+
+    private static final byte[] nonBase64 = {
```

Comment:
Please add a comment describing this test data.
(It looks like it could be generated more compactly than an explicit array).
Ditto nonBase64URL below.

Reply:
Will do.